src/python/pants/core/goals/tailor.py

```diff
@@ -596,7 +596,7 @@ async def tailor(
 
     if global_options.use_deprecated_directory_cli_args_semantics:
         search_paths = PutativeTargetsSearchPaths(
-            dirs=(), deprecated_recursive_dirs=specs_to_dirs(specs)
+            dirs=(), deprecated_recursive_dirs=specs_to_dirs(specs.includes)
```

Comment:
I don't see how the `includes` have been filtered by the `excludes`..?

Edit: ah, and seeing the specs rules, that is slated for a follow up PR. However, the ignores seems to be implemented in rules, so any code path operating on `Specs` directly, would have a hard time respecting the `ignores`.

Reply:
Ah so this call-site is unlike any other. `tailor.py` has its own custom handling of `Specs`, which is now deprecated. It used to treat `tailor dir` as == `tailor dir::`. Hence this helper function `specs_to_dirs`, which will be deleted.

If you are using the deprecated CLI semantics, then ignores won't work specifically with the `tailor` goal. That is only out of laziness, that I did not want to figure out how to get ignores working with `specs_to_dirs`. If you want the latest and greatest, you need to use the new semantics.

--

> However, the ignores seems to be implemented in rules, so any code path operating on Specs directly, would have a hard time respecting the ignores.

To be clear, all other call-sites will not need to worry about includes vs. ignores. They simply use `Get(Addresses, Specs)` and Pants will do the right thing.